src/hotspot/os/linux/os_linux.cpp

```diff
@@ -5826,13 +5826,27 @@ static bool compute_crengine() {
 }
 
 static int call_crengine() {
-  if (_crengine && !fork()) {
+  if (!_crengine) {
+    return -1;
+  }
+
+  pid_t pid = fork();
```

Comment:
Should we check for fork() to fail?  If so, we need to handle `-1` separately from `0` (child) and `> 0` (parent)